auth/api/v1/api.go

```diff
@@ -257,26 +257,72 @@ func (w Wrapper) DrawUpContract(ctx echo.Context) error {
 	return ctx.JSON(http.StatusOK, response)
 }
 
-// CreateJwtBearerToken handles the http request (from from the vendor's EPD/XIS) for creating a JWT bearer token which can be used to retrieve an access token from a remote Nuts node.
-func (w Wrapper) CreateJwtBearerToken(ctx echo.Context) error {
-	requestBody := &CreateJwtBearerTokenRequest{}
+// CreateJwtGrant handles the http request (from from the vendor's EPD/XIS) for creating a JWT bearer token which can be used to retrieve an access token from a remote Nuts node.
+func (w Wrapper) CreateJwtGrant(ctx echo.Context) error {
+	requestBody := &CreateJwtGrantRequest{}
 	if err := ctx.Bind(requestBody); err != nil {
 		return err
 	}
 
-	request := services.CreateJwtBearerTokenRequest{
+	request := services.CreateJwtGrantRequest{
 		Actor:         requestBody.Actor,
 		Custodian:     requestBody.Custodian,
 		IdentityToken: &requestBody.Identity,
 		Service:       requestBody.Service,
 		Subject:       requestBody.Subject,
 	}
-	response, err := w.Auth.OAuthClient().CreateJwtBearerToken(request)
+	response, err := w.Auth.OAuthClient().CreateJwtGrant(request)
 	if err != nil {
 		return core.InvalidInputError(err.Error())
 	}
 
-	return ctx.JSON(http.StatusOK, JwtBearerTokenResponse{BearerToken: response.BearerToken})
+	return ctx.JSON(http.StatusOK, JwtGrantResponse{BearerToken: response.BearerToken})
+}
+
+// RequestAccessToken handles the HTTP request (from the vendor's EPD/XIS) for creating a JWT grant and using it as authorization grant to get an access token from the remote Nuts node.
+func (w Wrapper) RequestAccessToken(ctx echo.Context) error {
+	requestBody := &RequestAccessTokenRequest{}
+	if err := ctx.Bind(requestBody); err != nil {
+		return err
+	}
+
+	request := services.CreateJwtGrantRequest{
+		Actor:         requestBody.Actor,
+		Custodian:     requestBody.Custodian,
+		IdentityToken: &requestBody.Identity,
+		Service:       requestBody.Service,
+		Subject:       requestBody.Subject,
+	}
+
+	jwtGrantResponse, err := w.Auth.OAuthClient().CreateJwtGrant(request)
+	if err != nil {
+		logging.Log().WithError(err).Warn("Failed to create JWT grant")
+		return core.InvalidInputError(err.Error())
+	}
+
+	custodianDID, err := did.ParseDID(requestBody.Custodian)
+	if err != nil {
+		logging.Log().WithError(err).WithField("did", requestBody.Custodian).Warn("Invalid custodian DID")
+		return core.InvalidInputError(err.Error())
+	}
+
+	endpointURL, err := w.Auth.OAuthClient().GetOAuthEndpointURL(requestBody.Service, *custodianDID)
+	if err != nil {
+		logging.Log().WithError(err).Error("Failed to get OAuth Endpoint URL")
+		return fmt.Errorf("unable to find the oauth2 service endpoint of the custodian: %w", err)
```

Comment:
I agree. I think `412` would be the best option as we're not sure if this is actually the clients fault. In most cases it will be but it could also be an error in the vdr store.

Reply:
Fixed